### PR TITLE
[patterns] Enable accepting versioned components from client side

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Signed-off-by: Abhishek-kumar09 <abhimait1909@gmail.com>

**Description**
Store version details of components inside Service Struct

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: models/pattern/core/pattern.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ type Service struct {
 	Name        string            `yaml:"name,omitempty" json:"name,omitempty"`
 	Type        string            `yaml:"type,omitempty" json:"type,omitempty"`
 	Namespace   string            `yaml:"namespace" json:"namespace"`
+	Version     string            `yaml:"version" json:"version"`
```
Comment (leecalcote) [inline models/pattern/core/pattern.go:45]: @Abhishek-kumar09 is this the pattern version or other? How can we make this more clear?

Comment (Abhishek-kumar09) [reply models/pattern/core/pattern.go:45]: This is a Service Struct, thus it defines the service (Components in Extensions) version.

**Review by leecalcote — APPROVED** [review]

**Comment by leecalcote** [comment]
Looks good. There is a merge conflict to address, @Abhishek-kumar09

**Review by leecalcote — APPROVED** [review]

**Comment by leecalcote** [comment]
merge conflicts...

**Review by leecalcote — CHANGES_REQUESTED** [review]
Merge conflicts, still.

What happens if a component doesn't define a version?

**Comment by Abhishek-kumar09** [comment]
I components doesn't define a version, as in the case with all the designs until now, the components should be choosen from the latest version available.

**Comment by leecalcote** [comment]
That behavior sounds good, @Abhishek-kumar09

**Comment by Abhishek-kumar09** [comment]
@Revolyssup has to confirm for anti-patterns if avaiable.

**Comment by Revolyssup** [comment]
@Abhishek-kumar09  merge conflicts

**Comment by leecalcote** [comment]
@sudo-NithishKarthik be aware of this.

**Pull request merged**